docs/book/v2/custom-responses.md

```diff
@@ -1,260 +1,289 @@
-    {
-        // - Do something with $data, and create a Stream for the body (if necessary).
-        // - Maybe set some default headers.
-
-        parent::__construct($body, $status, $headers);
-    }
-}
-```
-
-Note the call to `parent::__construct()`. This is particularly relevant, as the implementation at
-the time of writing has all class properties marked as private, making them inaccessible to
-extensions; this is done to protect encapsulation and ensure consistency of operations between
-instances.
-
-If you don't want to go the extension route (perhaps you don't want another `ResponseInterface`
-implementation within your object graph) you can instead create a factory. As an example:
-
-```php
-$plainTextResponse = function ($text, $status = 200, array $headers = []) {
-    $response = new Zend\Diactoros\Response('php://temp', $status, $headers);
-    $response->getBody()->write($text);
-    if (! $response->hasHeader('Content-Type')) {
-        $response = $response->withHeader('Content-Type', 'text/plain');
-    }
-    return $response;
-};
-
-$response = $plainTextResponse('Hello, world!');
-```
-
-We recommend following the semantic of providing the status and headers as the final two arguments
-for any factory or custom response extensions.
+# Custom Responses
+
+When developing server-side applications, the message type you're most likely to create manually is
+the response. In such cases, the standard signature can be an obstacle to usability. Let's review:
+
+```php
+namespace Zend\Diactoros;
+
+use Psr\Http\Message\ResponseInterface;
+
+class Response implements ResponseInterface
+{
+    public function __construct($body = 'php://temp', $status = 200, array $headers = []);
+}
+```
+
+Some standard use cases, however, make this un-wieldy:
+
+- Returning a response containing HTML; in this case, you likely want to provide the HTML to the
+  constructor, not a stream with the HTML injected.
+- Returning a response containing JSON; in this case, you likely want to provide the data to
+  serialize to JSON, not a stream containing serialized JSON.
+- Returning a response with no content; in this case, you don't want to bother with the body at all.
+- Returning a redirect response; in this case, you likely just want to specify the target for the
+  `Location` header, and optionally the status code.
+
+Starting with version 1.1, Diactoros offers several custom response types for simplifying these
+common tasks.
+
+## Text Responses
+
+`Zend\Diactoros\Response\TextResponse` creates a plain text response. It sets the
+`Content-Type` header to `text/plain` by default:
+
+```php
+$response = new Zend\Diactoros\Response\TextResponse('Hello world!');
+```
+
+The constructor accepts two additional arguments: a status code and an array of headers.
+
+```php
+$response = new Zend\Diactoros\Response\TextResponse(
+    $text,
+    200,
+    ['Content-Type' => ['text/csv']]

```

Comment:
With the new response type for CSV, the media type for this example should be changed. Maybe `text/example´. (see [RFC 4735](https://tools.ietf.org/html/rfc4735))

Reply:
Thanks for the notes, @froschdesign. I'm not completely familiar with this, so am happy to go with your suggestion.